validation/src/static/scripts/index.ts

```diff
@@ -1 +1,178 @@
-console.log('Hello world!');
+// Helper function for shorter 'document.getElementById(id)'
+const getElem = (id: string) => document.getElementById(id);
+
+let editor: CodeMirror.EditorFromTextArea;
+
+window.onload = async () => {
+  // input textarea must be visible to set CodeMirror text editor up.
+  const snippetComparison = getElem('snippetComparison');
+  if (!snippetComparison) return;
+  snippetComparison.classList.add('visible');
+
+  const snippetInput = getElem('snippetInput') as HTMLTextAreaElement;
+  if (!snippetInput) return;
+  editor = CodeMirror.fromTextArea(snippetInput, {
+    lineNumbers: true,
+    lineWrapping: true,
+    mode: 'xml',
+  });
+  snippetComparison.classList.remove('visible');
+
+  // Get 'Preview' to display under 'Comparison History'
+  const rawResponse = await fetch('http://localhost:3000/api/preview');
+  const preview = (await rawResponse.json()) as Preview;
+  displayPreview(preview);
+};
+
+/**
+ * Sends call to run a snippet comparison on provided input.
+ * Displays results of that comparison.
+ */
+const getSnippetComparison = async () => {
+  const resultsTable = getElem('resultsTable');
+  const resultsContainer = getElem('snippetResults');
+  if (!resultsTable || !resultsContainer || !editor) return;
+
+  resultsTable.classList.add('hidden');
+  resultsContainer.innerHTML = '';
+
+  const rawResponse = await fetch(
+    'http://localhost:3000/api/runSnippetComparison',
+    {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({snippet: editor.getValue()}),
+    }
+  );
+
+  // Display results in table if response is OK
+  if (rawResponse.status === 200) {
+    const comparisonResults = await rawResponse.json();
+    const accnames = comparisonResults[0];
+
+    getElem('chromeAccName')!.innerText = '"' + accnames.chrome + '"';
+    getElem('axeAccName')!.innerText = '"' + accnames.axe + '"';
+    getElem('aomAccName')!.innerText = '"' + accnames.aom + '"';
+    getElem('bgAccName')!.innerText = '"' + accnames.bg + '"';
+    getElem('ourAccName')!.innerText = '"' + accnames.accname + '"';
+
+    resultsTable.classList.remove('hidden');
+
+    if (comparisonResults[1]) {
+      const caseId = comparisonResults[1];
+
+      resultsContainer.innerHTML += `
+        <div class="bluetext comparisonResultText">A test-case was generated as a result of this comparison - <a class="linkBtn" href="/case/${caseId}">View Test-case ${caseId}</a></div>
+      `;
+    }
+  } else if (rawResponse.status === 400) {
```

Comment:
When I see `else if` I expect that to mean that in some cases neither condition will be true and the flow will fall through. But that's not true here, the server should always return 200 or 400. I'd recommend changing this bit to a plain `else` block (applies throughout when these are the only options)